src/cli/commands/force/lightning/local/device/create.ts

```diff
@@ -150,27 +150,30 @@ export class Create extends Setup {
     }
 
     private async executeAndroidDeviceCreate(): Promise<any> {
-        const preferredPack = await AndroidSDKUtils.findRequiredEmulatorImages();
-        const emuImage = preferredPack.platformEmulatorImage || 'default';
-        const androidApi = preferredPack.platformAPI;
-        const abi = preferredPack.abi;
-
-        return AndroidSDKUtils.createNewVirtualDevice(
-            this.deviceName,
-            emuImage,
-            androidApi,
-            this.deviceType,
-            abi
+        return AndroidSDKUtils.findRequiredEmulatorImages().then(
```

Comment:
For the record, I don't know that I have a preference *across the board* for this pattern over the other. I haven't yet developed a keen sense of when to use one over the other—some obvious cases (to me) are e.g. when there are async blocks nested 10 deep and as such increasingly hard to follow, or complex conditional async calls that might otherwise benefit from a "flattening". Other cases aren't so obvious to me.

To be clear, I think the change you've made here is fine. I just figured I'd throw out the follow-on thought processes I engaged around reviewing it.

Reply:
I'm glad you brought this up, b/c I thought of it too and then thought to myself that, if we ever have a function that is going to be that complex, then that would be our cue to break up the function into smaller functions. But yes, the idea of using chaining vs "flattened" approach should not be set in stone and we should evaluate on as needed basis.

Reply:
Inline `async` [here](https://github.com/forcedotcom/lwc-dev-mobile/blob/c14c1dc9539205405143f77e104373a1392c9e5b/src/common/AndroidUtils.ts#L187) I thought was unique.

About to finish reviewing AndroidUtil.ts. That's a long one 😌